File: DotNet/Models/InvoiceHeader.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Vehicle_Configurator_Project.Models
{
    [Table("invoice_header")]
    public class InvoiceHeader
    {
        [Key]
        [Column("inv_id")]
        public int InvId { get; set; }

        // Many-to-One relationship with User
        [Required]
        [ForeignKey("UserId")]
        [JsonIgnore]
        public User User { get; set; } = null!;
        [Column("userid")]
        public int UserId { get; set; }

        [Required]
        [Column("inv_date", TypeName = "datetime")]
        public DateTime InvDate { get; set; } = DateTime.Now;

        // Many-to-One with Model (lazy loading can be configured in DbContext)
        [Required]
        [ForeignKey("ModelId")]
        [JsonIgnore]
        public Model Model { get; set; } = null!;
        [Column("model_id")]
        public int ModelId { get; set; }

        // One-to-Many relationship with InvoiceDetail
        [JsonIgnore]
        public List<InvoiceDetail> InvoiceDetails { get; set; } = new List<InvoiceDetail>();

        [Required]
        [Column("quantity")]
        public int Quantity { get; set; }

        [Required]
        [Column("total_amount", TypeName = "decimal(12,2)")]
        public decimal TotalAmount { get; set; }

        [Required]
        [Column("tax", TypeName = "decimal(12,2)")]
        public decimal Tax { get; set; }

        [Required]
        [Column("final_amount", TypeName = "decimal(12,2)")]
        public decimal FinalAmount { get; set; }

        [Required]
        [Column("cust_details", TypeName = "nvarchar(500)")]
        public string CustDetails { get; set; } = null!;
    }
}

File: JAVA/VehicleConfigurator/src/main/java/com/example/entities/InvoiceHeader.java
package com.example.entities;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "invoiceDetails", "user"})
@Table(name = "invoice_header")
public class InvoiceHeader {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="inv_id")
    private int invId;
    
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "userid", nullable = false)
    private User user;
    
	@Column(name = "inv_date", columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP", nullable = false)
    private LocalDateTime invDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "model_id", nullable = false)
    private Model model;
    
    @OneToMany(mappedBy = "invoiceHeader", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<InvoiceDetail> invoiceDetails;
    
    @Column(name = "quantity", nullable = false)
    private int quantity;

    @Column(name = "total_amount", precision = 12, scale = 2, nullable = false)
    private BigDecimal totalAmount;

    @Column(name = "tax", precision = 12, scale = 2, nullable = false)
    private BigDecimal tax;

    @Column(name = "final_amount", precision = 12, scale = 2, nullable = false)
    private BigDecimal finalAmount;

    @Column(name = "cust_details", length = 500, nullable = false)
    private String custDetails;

    public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

    public int getInvId() {
		return invId;
	}

	public void setInvId(int invId) {
		this.invId = invId;
	}

	public LocalDateTime getInvDate() {
		return invDate;
	}

	public void setInvDate(LocalDateTime invDate) {
		this.invDate = invDate;
	}

	public Model getModel() {
		return model;
	}

	public void setModel(Model model) {
		this.model = model;
	}

	public int getQuantity() {
		return quantity;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

	public BigDecimal getTotalAmount() {
		return totalAmount;
	}

	public void setTotalAmount(BigDecimal totalAmount) {
		this.totalAmount = totalAmount;
	}

	public BigDecimal getTax() {
		return tax;
	}

	public void setTax(BigDecimal tax) {
		this.tax = tax;
	}

	public BigDecimal getFinalAmount() {
		return finalAmount;
	}

	public void setFinalAmount(BigDecimal finalAmount) {
		this.finalAmount = finalAmount;
	}

	public String getCustDetails() {
		return custDetails;
	}

	public void setCustDetails(String custDetails) {
		this.custDetails = custDetails;
	}
	
	public List<InvoiceDetail> getInvoiceDetails() { return invoiceDetails; }

    public void setInvoiceDetails(List<InvoiceDetail> invoiceDetails) {
        this.invoiceDetails = invoiceDetails;
    }

}
